### PR TITLE
[Elastic Agent] Add support for Fleet Server inside Docker

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

First it refactors the entrypoint for the Elastic Agent docker container to use a new subcommand `container`. This subcommand is designed specifically to be executed by the container runtime. Based on the defined environment variables it prepares the Elastic Agent to run in that environment. Due to the large number of environment variables that Elastic Agent needed and the complexity of the order of operations based it was best to move this from a bash script to golang.

With the refactor comes the ability to bootstrap the Elastic Agent inside of a docker container. Using the `FLEET_SERVER_ENABLE` flag the docker container will bootstrap the Fleet Server and enroll the Elastic Agent all inside the container on startup.

## Why is it important?

To allow Fleet Server to be spawned under Elastic Agent while being executed under Docker.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/cmd/container.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+* Elastic Agent Fleet Enrollment
+  This enrolls the Elastic Agent into a Fleet Server. It is also possible to have this create a new enrollment token
+  for this specific Elastic Agent.
+
+  FLEET_ENROLL - set to 1 for enrollment to occur
+  FLEET_URL - URL of the Fleet Server to enroll into
+  FLEET_ENROLLMENT_TOKEN - token to use for enrollment
+  FLEET_TOKEN_NAME - token name to use for fetching token from Kibana
+  FLEET_TOKEN_POLICY_NAME - token policy name to use for fetching token from Kibana
+  FLEET_INSECURE - communicate with Fleet with either insecure HTTP or un-verified HTTPS
```
Comment (simitt) [inline x-pack/elastic-agent/pkg/agent/cmd/container.go:76]: This section mixes `Fleet` and `Fleet Server` - shouldn't the ENVs also be `FLEET_SERVER_xxx`?

Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:76]: These are for the Agent itself to enroll. `FLEET_INSECURE` is different then `FLEET_SERVER_INSECURE_HTTP`.

Comment (simitt) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:76]: Should they be `KIBANA_FLEET_xyz` then instead? In the other ENVs you clearly separate between `FLEET_SERVER` and `KIBANA_FLEET`.

Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:76]: These are not for Kibana, but are for the Agent itself. Just like the `FLEET_ENROLLMENT_TOKEN`. For the Agent its only `FLEET_*`.

**Review by simitt — APPROVED** [review]
Changes LGTM

**Pull request merged**